File: buerguer-pig/AMarotoChivite-BurguerPig/AMarotoChivite-Aemet-Java/src/main/java/Ficheros/Aemet/controllers/ControllerPrincipal.java
package Ficheros.Aemet.controllers;

import Ficheros.Aemet.models.Aemet;
import Ficheros.Aemet.repositories.RepositoryToWriteRead;
import ch.qos.logback.classic.Logger;
import org.slf4j.LoggerFactory;

import java.util.List;

public class ControllerPrincipal {

    private final RepositoryToWriteRead repositoryToWriteRead;
    private final Logger logger = (Logger) LoggerFactory.getLogger(ControllerPrincipal.class);

    public ControllerPrincipal(RepositoryToWriteRead repositoryToWriteRead) {
        this.repositoryToWriteRead = repositoryToWriteRead;
    }

    public void saveInRepository(List<Aemet> list) {
        logger.debug("Controller: Guardando items en repository");
        repositoryToWriteRead.saveItems(list);
    }

    public void saveInFile() {
        logger.debug("Controller: Escribiendo (sobreescribiendo)");
        repositoryToWriteRead.saveFile();
    }

    public List<Aemet> readInFile() {
        logger.debug("Controller: Leyendo");
        return repositoryToWriteRead.readFile();
    }

}

File: aemet/AMarotoChivite-Aemet/AMarotoChivite-Aemet-Kotlin/src/main/kotlin/Ficheros/Aemet/controllers/ControllerPrincipal.kt
package Ficheros.Aemet.controllers

import Ficheros.Aemet.models.Aemet
import Ficheros.Aemet.repositories.RepositoryToImportExport
import mu.KotlinLogging

class ControllerPrincipal(private val repositoryToExportImport: RepositoryToImportExport) {

    private val logger = KotlinLogging.logger {}

    fun saveInRepository(list: List<Aemet>) {
        repositoryToExportImport.saveItems(list)
    }

    fun saveInFile() {
        logger.debug { "Controller: Escribiendo (sobreescribiendo)" }
        repositoryToExportImport.saveAllModelsInFileOverride()
    }

    fun readInFile(): List<Aemet> {
        logger.debug { "Controller: Leyendo" }
        return repositoryToExportImport.readAllModelsInFile()
    }

}